### ISSUE TITLE
ensurepip bootstrap breaks out of isolated environment

### ISSUE THREAD

**Issue by 75e79e7a-f91c-4ae5-b0c9-c13e306f78ff** [python/cpython#90355]
BPO | [46197](https://bugs.python.org/issue46197)
--- | :---
Nosy | @gvanrossum, @ncoghlan, @merwok, @ericsnowcurrently, @dstufft, @pradyunsg, @cdce8p, @kcdodd
PRs | <li>python/cpython#30307</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-12-29.23:24:11.769>
labels = ['type-bug', '3.8', '3.9', '3.10', '3.11']
title = 'ensurepip bootstrap breaks out of isolated environment'
updated_at = <Date 2022-03-29.20:57:30.348>
user = 'https://github.com/kcdodd'
```

bugs.python.org fields:
```python
activity = <Date 2022-03-29.20:57:30.348>
actor = 'gvanrossum'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = []
creation = <Date 2021-12-29.23:24:11.769>
creator = 'kcdodd'
dependencies = []
files = []
hgrepos = []
issue_num = 46197
keywords = ['patch']
message_count = 2.0
messages = ['409334', '412408']
nosy_count = 8.0
nosy_names = ['gvanrossum', 'ncoghlan', 'eric.araujo', 'eric.snow', 'dstufft', 'pradyunsg', 'cdce8p', 'kcdodd']
pr_nums = ['30307']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue46197'
versions = ['Python 3.8', 'Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

**Comment by gvanrossum** [comment]
@kcdodd Pleas follow instructions here.

**Comment by kcdodd** [comment]
Should the pull request be to "main", or to "3.11"?

**Comment by gvanrossum** [comment]
It should be main (that comment was written when 3.11==main, but now 3.11b1 was released and 3.12==main until next year). I'll look into the CLA bot.

**Comment by ambv** [comment]
This is now fixed in 3.8 - 3.12. Thanks, Carter! ✨ 🍰 ✨
